### PR TITLE
Inserts additional information about the relationship lifecycle into layers-interfaces Splits the interfaces doc into 2 pages for theory => implementation concepts

### PR DESCRIPTION
**Pull request by lazypower**
WIP - probably needs additional love

### CODE REVIEW

File: src/en/developer-layers-interfaces.md [hunk 1]
```diff
@@ -39,230 +39,47 @@ this interface?
...
 
-#### GLOBAL
+### Relation Scopes
 
-All connected services and units for this relation will share a single
-conversation. The same data will be broadcast to every remote unit, and
-retrieved data will be aggregated across all remote units and is expected to
-either eventually agree or be set by a single leader.
+An interfaces relationship scope falls into one of two categories. An implied
+`global` scope, which is the default for all interfaces. Or a `container` scope
```
Comment (mbruzek) [inline src/en/developer-layers-interfaces.md]: Capitalize the relationship scope names here to match what is in the next document (GLOBAL, CONTAINER).  These would also be great links to the next document with the appropriate anchors.

File: src/en/developer-layers-interfaces.md [hunk 2]
```diff
@@ -39,230 +39,47 @@ this interface?
...
+which is explicitly defined. That is to say: a globally-scoped relation has a
+single unit scope, whilst a container-scoped relation has one for each principal
+unit.
 
-```python
-class MyRelationClient(RelationBase):
-  scope = Scopes.GLOBAL
-```
+You will typically only require a `container` scoped relation when programming
```
Comment (mbruzek) [inline src/en/developer-layers-interfaces.md]: The scopes listed in developer-layers-interfaces2.md are GLOBAL, SERVICE, and UNIT, where does `container` come from?

File: src/en/developer-layers-interfaces2.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  website:
+    interface: http
+```
+
+And in the reactive class, we will receive the RelationBase object, and invoke
+the configure method:
+
+```python
+@when('website.available')
+def configure_website(website):
```
Comment (mbruzek) [inline src/en/developer-layers-interfaces2.md]: It always confuses me where the argument (website) comes from. You have explained it a few times, but it is worth calling out "website" in this example.

**Comment by mbruzek** [comment]
Had a few comments about the use of "container".

File: src/en/developer-layers-interfaces.md (hunk 1 above)
Comment (chuckbutler) [inline src/en/developer-layers-interfaces.md]: This is a byproduct of the overloaded namespaces with scope in relations, and scope when defining interface layers. As both topics arise in the same area of analysis i've added additional context here to attempt to disambiguate the terms.

File: src/en/developer-layers-interfaces.md (hunk 2 above)
Comment (chuckbutler) [inline src/en/developer-layers-interfaces.md]: From programming relationships when you're using a subordinate. 

see: https://jujucharms.com/docs/1.25/authors-subordinate-services  subheading "Container Relation"

We still need subordinates represented in our docs per #10

**Comment by chuckbutler** [comment]
@mbruzek thats an unfortunate side effect of the overloaded / overlapping terms of scope when used in interface layers vs relationship declarations in metadata.yaml. I've tried to clear this up in a forthcoming commit.

File: src/en/developer-layers-interfaces2.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+  website.configure(80)
+```
+
+Note that you see an implicit object being passed in as `website`. This is a
+behavior pattern when using Interface Layers. As we programmed the configure
+method above in the HTTP Interface, this relation (as defined in [`metadata.yaml`]())
```
Comment (mbruzek) [inline src/en/developer-layers-interfaces2.md]: Empty link here.  Should be authors-charm-metadata.html

File: src/en/developer-layers-interfaces2.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+thats implementing The interface is *website*. Thus we get an instance of the HTTP interface, as
+the website object, which we can then invoke the `configure()`` method defined above
```
Comment (mbruzek) [inline src/en/developer-layers-interfaces2.md]: extra backtick here.

File: src/en/reference-metadata.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
+Title: Metadata.yaml reference
+
+The only file that must be present in a charm is `metadata.yaml`, in the root
+directory. A metadata file must be a valid yaml dictionary, containing at least
+the following fields:
+
+### name
+is the charm name, which is used to form the charm URL.
+M
```
Comment (mbruzek) [inline src/en/reference-metadata.md]: M?

Comment (chuckbutler) [inline src/en/reference-metadata.md]: M :cop:

File: src/en/developer-layers-interfaces2.md (hunk 3 above)
Comment (chuckbutler) [inline src/en/developer-layers-interfaces2.md]: ambigous and seemingly arbitrary object reference :cop:

File: src/en/reference-metadata.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+- network
+- storage
+- database
+- cache-proxy
+- application_development
+- web_server
+
+
+### subordinate
+boolean value denoting if the charm is a [`subordinate`](#TODO) if the charm is
```
Comment (mbruzek) [inline src/en/reference-metadata.md]: Issue #10 to fix this link.

**Comment by mbruzek** [comment]
Finally this gets my approval!  About time.  Signed  grammar :cop:

**Comment by chuckbutler** [comment]
:no_bicycles:

**Pull request merged**